### PR TITLE
[Chore] Fix production CI failure

### PR DESCRIPTION
**Pull request by markgravity**
## What happened

- Restore lost Production CI configuration
- Update newest Fastlane from ios-template

## Insight

n/a

## Proof Of Work

Screen Shot 2021-09-15 at 16 21 21

### CODE REVIEW

**Review by minhnimble — COMMENTED** [review]
there is one small comment, the rest LGTM 🚀

File: .github/workflows/deploy_testflight.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Deploy to Test Flight (production)
 
 on:
   push:
-    branches: [ master, main ]
+     branches: [ master, main ]
```
Comment (minhnimble) [inline .github/workflows/deploy_testflight.yml:5]: please revert this change

**Review by minhnimble — APPROVED** [review]
LGTM 🚀

File: fastlane/Constants/Constants.rb [hunk 2]
```diff
@@ -82,10 +92,25 @@ def self.FIREBASE_APP_ID_PRODUCTION
...
+
+  # Keychain name
+  def self.KEYCHAIN_NAME
+    'github_action_keychain'
+  end
+
+  # a scheme name for unit testing
+  def self.KEYCHAIN_PASSWORD
+    'password'
+  end
```
Comment (suho) [inline fastlane/Constants/Constants.rb:112]: Please remove this out of the Constants because this is a password, we should define a secret for this

File: fastlane/Fastfile [hunk 3]
```diff
@@ -1,36 +1,38 @@
...
-  firebase_token: ENV['FIREBASE_CLI_TOKEN']
+  firebase_token: Environments.FIREBASE_CLI_TOKEN
 )
 
-keychain_name = "github_action_keychain"
-keychain_password = "password"
+match_manager = MatchManager.new(
+  fastlane: self,
+  keychain_name: Constants.KEYCHAIN_NAME,
+  keychain_password: Constants.KEYCHAIN_PASSWORD,
```
Comment (suho) [inline fastlane/Fastfile:33]:
```suggestion
  keychain_password: Environments.KEYCHAIN_PASSWORD,
```

**Review by suho — APPROVED** [review]
